mypy.ini

```diff
@@ -16,7 +16,13 @@ disallow_untyped_defs = true
 [mypy-homeassistant.config_entries]
 disallow_untyped_defs = false
 
+[mypy-homeassistant.components.switcher_kis.*]
```

Comment:
Out of the 100s of components and platforms, not a single one has needed to update this file.  Please fix the design issues in the code or add local ignores and undo these changes.

Reply:
@andrewsayre 
> Out of the 100s of components and platforms, not a single one has needed to update this file.

I don't think `mypy` is officially maintained yet from the HA development checklist point-of-view.
I think the reason for that file not being updated by anyone is probably because there is no `mypy` mandatory stage in the development checklist yet (but I'm sure there will be one soon).
Anyway, if you run:
```shell
mypy homeassistant/ tests/
```
within in your HA development environment, you'll see a ton of mypy errors,
but none of them will be from my`switcher_kis` component, because I updated this file.
:-)

My problem here was the following functions aren't static typed and needed to called from within a typed context:
- *homeassistant.helpers.discovery.async_load_platform*
- *homeassistant.helpers.event.async_call_later*
(will be replaced as you suggested in another change request)
- *homeassistant.helpers.entity.Entity.async_schedule_update_ha_state*

Once those functions will be updated with the static typing syntax,
My additions to the `mypy.ini` file can be deleted.

If there's a problem with that, I'll be happy to delete these additions.
But there's no actual redesigning required on the component end,
these errors I tried to suppress here will just join the list of errors from other components.

Please advice on what would you recommend me to do next.
Thank you.
:-)

Reply:
BTW
The errors I tried to suppress are:
```text

homeassistant/components/switcher_kis/__init__.py:90: error: Call to untyped function "async_load_platform" in typed context
homeassistant/components/switcher_kis/__init__.py:100: error: Call to untyped function "async_call_later" in typed context
homeassistant/components/switcher_kis/__init__.py:103: error: Call to untyped function "async_call_later" in typed context
homeassistant/components/switcher_kis/switch.py:176: error: Call to untyped function "async_schedule_update_ha_state" of "Entity" in typed context
homeassistant/components/switcher_kis/switch.py:202: error: Call to untyped function "async_schedule_update_ha_state" of "Entity" in typed context
homeassistant/components/switcher_kis/switch.py:228: error: Call to untyped function "async_schedule_update_ha_state" of "Entity" in typed context
```

I think I can get rid of the `tests` addition, I'm just now noticing there is no error poping from the tests path after the last commit.

Reply:
Please revert this change. We have not turn on mypy on components, it would not happen anytime soon. The big road lock is we don't install requirements lib when we run mypy on CI.

I also suggest you look at esphome component, it used typings as well. I like the way esphome deal with the import only needed for typings purpose.

Reply:
All functions in helpers should be static typed, if they are not, I will prefer add static check for it first. Of course, in a separate pull request.

Reply:
@awarecan 
Got it, thanks, I'll revert this change.
If there is any plans on static typing parts of  the HA that you are aware or part of,
I would love to take part and do some of the work.

Reply:
@awarecan 
BTW
I've looked at the `esphome` component.
Turns out I've totally missed `typing.TYPE_CHECKING` constant, that is a game changer.
;-)

Reply:
Fixed in this [commit](https://github.com/home-assistant/home-assistant/pull/22325/commits/ba9ccb5f70f0807dfbf1b084703d8665edf69ffa).